server/src/main/java/org/elasticsearch/common/lucene/search/function/ScriptScoreQuery.java

```diff
@@ -86,9 +89,18 @@ public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float bo
             return subQuery.createWeight(searcher, scoreMode, boost);
```

Comment:
when is this branch executed? and do we still need the `needsTermStatistics` check here and do `this.visit(QueryVisitor.termCollector(terms));`?